velox/exec/PrefixSort.cpp

```diff
@@ -185,7 +216,12 @@ int PrefixSort::comparePartNormalizedKeys(char* left, char* right) {
   // If prefixes are equal, compare the remaining sort keys with rowContainer.
   char* leftRow = getRowAddrFromPrefixBuffer(left);
   char* rightRow = getRowAddrFromPrefixBuffer(right);
-  for (auto i = sortLayout_.numNormalizedKeys; i < sortLayout_.numKeys; ++i) {
+  // Since the last key is only partially encoded in the prefix, we must start
+  // the comparison from numNormalizedKeys - 1， otherwise numNormalizedKeys.
+  const auto start = sortLayout_.hasPartialNormalizedKey
```

Comment:
Please move the start calculation to sortLayout member, and remove the `hasPartialNormalizedKey`. This comparison is for each row, this will make it a bit faster.